### PR TITLE
Zadanie 2 wykonane

### PR DESCRIPTION
**Pull request by francesspot**

### CODE REVIEW

**Review by RakPiotr — CHANGES_REQUESTED** [review]
Tutaj: (przy wyborze metryk)
`scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)`
użyłeś standard scalera na CAŁYCH danych X_train. To oznacza że teraz każda dana zawiera w sobie informację o całym X_train:
`x_scaled = (x - μ) / σ`
(każda informacja zawiera informację o średniej całej kolumny, i odchyleniu standardowym)
to oznacza że później gdy używasz grid searchu:
`knn_grid_search.fit(X_train_scaled, y_train)`
przy walidacji krzyżowej, statystyki użyte do skalowania pochodzą również z przykładów, które w poszczególnym foldzie pełnią funkcję danych walidacyjnych. Informacje z części walidacyjnej wpływają więc na dane treningowe, co stanowi wyciek danych podczas walidacji krzyżowej i może prowadzić do lekko zawyżonych wyników.
Aby temu zapobiec proponuję użycia pipeline
`knn_pipeline = Pipeline([
    ('scaler', StandardScaler()),
    ('knn', KNeighborsClassifier())
])`
Dzięki temu każdy fold będzie skalowany tylko do swoich danych

**Review by Karol-Rejek — COMMENTED** [review]
Mógłbyś trochę lepiej przedstawić swój sposób rozumowania w opisach. Super że poprawiłeś problem z KNN. Przeglądając pracę raczej nie zauważyłem kolejnych bardzo rażących błędów.

**Review by Karolina4033 — COMMENTED** [review]
Dobrze zrobione zadanie, przejrzyste wykresy, opisane co robisz :) Nie wiem jak uzupełniałeś dane w poprzednim zadaniu, ale pamiętaj, że jeśli używałeś do tego mediany, czy dominanty, to mogło dojść do wycieku danych, bo wartosci uzupełniałeś na podstawie wszystkich danych, przed podziałem na zbiory testowe i treningowe.

**Comment by Dominik-Galus** [comment]
Hej, spoko i zwięzły notebook. Dane dobrze podzielone, przy knn używasz standardscalera więc tutaj bangla, drzewo słusznie bez skalowania. wnioski dobrze odnoszą się do dummy'ego i do wyników modeli. Brakuje mi tutaj porównania train vs test, jest to w wymogu zadania, przydałoby się policzyć accuracy/classification_report, żeby opisać wyniki na treningowym i testowym. Imo fajnie jakby były bardziej rozbudowane wnioski o dlaczego np. dlaczego małe n_neighbors = wrażliwość na szum
